### ISSUE TITLE
CI workflows reset when setting 'benchmark' label

### ISSUE THREAD

**Issue by hanno-becker** [pq-code-package/mlkem-c-aarch64#213]
When opening a PR, the functional CI is run automatically, but benchmarking only when the 'benchmark' label is set. However, it seems that when setting that label, the functional CI is reset as well, even if it did already run. This seems to have changed with #183

**Comment by hanno-becker** [comment]
@potsrevennil Can you look into this?

**Comment by potsrevennil** [comment]
I don't think this is changed upon last pr though
this should exist already, we just didn't notice it before

And this is due to that CI is triggered by pr label event, since we want to control ci workflow by label `ci`, but we only check for this on `ec2_all` job, not on the rest of the jobs

```
on:
  pull_request:
    branches: ["main"]
    types: [ "labeled", "synchronize" ]
...
jobs:
...
  ec2_all:

    if: github.repository_owner == 'pq-code-package' && (github.event.label.name == 'ci' || !github.event.pull_request.head.repo.fork)
```

**Comment by hanno-becker** [comment]
Can you fire up a PR for that?

**Comment by hanno-becker** [comment]
Fixed in #217
